mlflow/langchain/langchain_tracer.py

```diff
@@ -0,0 +1,428 @@
+    input arguments added to original function call.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._mlflow_client = MlflowClient()
+        self._run_span_mapping: Dict[str, MlflowSpanWrapper] = {}
+
+    def _get_span_by_run_id(self, run_id: UUID) -> Optional[MlflowSpanWrapper]:
+        if span := self._run_span_mapping.get(str(run_id)):
+            return span
+        raise MlflowException(f"Span for run_id {run_id!s} not found.")
+
+    def _start_span(
+        self,
+        span_name: str,
+        parent_run_id: Optional[UUID],
+        span_type: str,
+        run_id: UUID,
+        inputs: Optional[Dict[str, Any]] = None,
+        attributes: Optional[Dict[str, Any]] = None,
+    ) -> MlflowSpanWrapper:
+        """Start MLflow Span (or Trace if it is root component)"""
+        parent = self._get_span_by_run_id(parent_run_id) if parent_run_id else None
+        if parent:
+            span = self._mlflow_client.start_span(
+                name=span_name,
+                request_id=parent.request_id,
+                parent_span_id=parent.span_id,
+                span_type=span_type,
+                inputs=inputs,
+                attributes=attributes,
+            )
+        else:
+            # When parent_run_id is None, this is root component so start trace
+            span = self._mlflow_client.start_trace(
+                name=span_name, span_type=span_type, inputs=inputs, attributes=attributes
+            )
+        self._run_span_mapping[str(run_id)] = span
+        return span
+
+    def _end_span(
+        self,
+        span: MlflowSpanWrapper,
+        outputs=None,
+        attributes=None,
+        status=SpanStatus(TraceStatus.OK),
+    ):
+        """Close MLflow Span (or Trace if it is root component)"""
+        self._mlflow_client.end_span(
+            request_id=span.request_id,
+            span_id=span.span_id,
+            outputs=outputs,
+            attributes=attributes,
+            status=status,
+        )
+
+    def _reset(self):
+        self._run_span_mapping = {}
+
+    @override
+    def on_chat_model_start(
+        self,
+        serialized: Dict[str, Any],
+        messages: List[List[BaseMessage]],
+        *,
+        run_id: UUID,
+        tags: Optional[List[str]] = None,
+        parent_run_id: Optional[UUID] = None,
+        metadata: Optional[Dict[str, Any]] = None,
+        name: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        """Run when a chat model starts running."""
+        if metadata:
+            kwargs.update({"metadata": metadata})
+        llm_inputs = {"messages": [[msg.dict() for msg in batch] for batch in messages]}
+        self._start_span(
+            span_name=name or "chat model",
+            parent_run_id=parent_run_id,
+            # we use LLM for chat models as well
+            span_type=SpanType.LLM,
+            run_id=run_id,
+            inputs=llm_inputs,
```

Comment:
```suggestion
            inputs=messages,
```

(works fine on my end so for simplicity)

Reply:
I'll wait for Daniel's fix and then test